Model/Api/CreateOrder.php

```diff
@@ -0,0 +1,694 @@
+     */
+    private $url;
+
+    /**
+     * @var BackendUrl
+     */
+    private $backendUrl;
+
+    /**
+     * @var StockRegistryInterface
+     */
+    private $stockRegistry;
+
+    /**
+     * @var SessionHelper
+     */
+    private $sessionHelper;
+
+    /**
+     * @param HookHelper             $hookHelper
+     * @param OrderHelper            $orderHelper
+     * @param CartHelper             $cartHelper
+     * @param LogHelper              $logHelper
+     * @param Request                $request
+     * @param Bugsnag                $bugsnag
+     * @param Response               $response
+     * @param UrlInterface           $url
+     * @param BackendUrl             $backendUrl
+     * @param ConfigHelper           $configHelper
+     * @param StockRegistryInterface $stockRegistry
+     * @param SessionHelper          $sessionHelper
+     */
+    public function __construct(
+        HookHelper $hookHelper,
+        OrderHelper $orderHelper,
+        CartHelper $cartHelper,
+        LogHelper $logHelper,
+        Request $request,
+        Bugsnag $bugsnag,
+        Response $response,
+        UrlInterface $url,
+        BackendUrl $backendUrl,
+        ConfigHelper $configHelper,
+        StockRegistryInterface $stockRegistry,
+        SessionHelper $sessionHelper
+    ) {
+        $this->hookHelper = $hookHelper;
+        $this->orderHelper = $orderHelper;
+        $this->logHelper = $logHelper;
+        $this->request = $request;
+        $this->bugsnag = $bugsnag;
+        $this->response = $response;
+        $this->configHelper = $configHelper;
+        $this->cartHelper = $cartHelper;
+        $this->url = $url;
+        $this->backendUrl = $backendUrl;
+        $this->stockRegistry = $stockRegistry;
+        $this->sessionHelper = $sessionHelper;
+    }
+
+    /**
+     * Pre-Auth hook: Create order.
+     *
+     * @api
+     *
+     * @param string $type - "order.create"
+     * @param array $order
+     * @param string $currency
+     *
+     * @return void
+     * @throws \Exception
+     */
+    public function execute(
+        $type = null,
+        $order = null,
+        $currency = null
+    ) {
+        try {
+            if ($type !== 'order.create') {
+                throw new BoltException(
+                    __('Invalid hook type!'),
+                    null,
+                    self::E_BOLT_GENERAL_ERROR
+                );
+            }
+
+            $payload = $this->request->getContent();
+
+            $this->validateHook();
+
+            if (empty($order)) {
+                throw new BoltException(
+                    __('Missing order data.'),
+                    null,
+                    self::E_BOLT_GENERAL_ERROR
+                );
+            }
+
+            $quoteId = $this->getQuoteIdFromPayloadOrder($order);
+            /** @var Quote $immutableQuote */
+            $immutableQuote = $this->loadQuoteData($quoteId);
+
+            // Convert to stdClass
```

Comment:
no need to comment, decent php developer knows what json_decode does

Reply:
done